server/src/main/java/org/opensearch/identity/AuthenticationManager.java

```diff
@@ -0,0 +1,61 @@
+package org.opensearch.identity;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Authentication management for OpenSearch.
+ *
+ * Retrieve the current subject, switch to a subject, and persist subject identity through request lifetime
+ * */
+public interface AuthenticationManager {
+
+    /**
+     * Get the current subject
+     * */
+    public Subject getCurrentSubject();
+
+    // Update the current subject
+
+    /**
+     * Authenticate a Subject via an http header
+     *
+     * Note: define exceptional states
+     * */
+    public void authenticateWithHeader(final String authorizationHeader);
+
+    /**
+     * authenticateAs a Subject by an identifier
+     *
+     * Why dangerous?  authenticateAs(...) allows OpenSearch to run as Subject without
+     * any other checks or safeguards.  If downstream code is not safeguarded it could run
+     * as permissions that are not intented, resulting in security issues.
+     *
+     * External data should never be directly populate subject
+     *
+     * Note: define exceptional states
+     *
+     * */
+    public AuthenticationSession dangerousAuthenticateAs(final String subject);
```

Comment:
So why even have this? Tells me something is broken by design.

Reply:
Revisited this - removed the `dangerous` api by not allowing this kind of action.  Moving the 'system' scenario into `OpenSearchThreadFactory` which has partitioning logic for different functions within OpenSearch (management/snapshot/system_write).